test/runtime_tests.cc

```diff
@@ -1013,5 +1015,58 @@ namespace detail {
 		dry_run_with_nodes(nodes);
 	}
 
+	TEST_CASE_METHOD(test_utils::mpi_fixture, "Config reads environment variables correctly", "[env-vars]") {
+		std::unordered_map<std::string, std::string> valid_test_env_vars{
+		    {"CELERITY_LOG_LEVEL", "debug"},
+		    {"CELERITY_GRAPH_PRINT_MAX_VERTS", "1"},
+		    {"CELERITY_DEVICES", "0 0"},
+		    {"CELERITY_PROFILE_KERNEL", "1"},
+		    {"CELERITY_DRY_RUN_NODES", "4"},
+		};
+
+		const auto test_env = env::scoped_test_environment(valid_test_env_vars);
+		auto cfg = config(nullptr, nullptr);
+
+		REQUIRE(spdlog::get_level() == spdlog::level::debug);
+		REQUIRE(cfg.get_graph_print_max_verts() == 1);
+		REQUIRE(config_testspy::get_device_list(cfg) == std::vector<size_t>{0, 0});
+		const auto has_prof = cfg.get_enable_device_profiling();
+		REQUIRE(has_prof.has_value());
+		REQUIRE((*has_prof) == true);
```

Comment:
There is no check inspecting that the number of dry run nodes was set correctly.